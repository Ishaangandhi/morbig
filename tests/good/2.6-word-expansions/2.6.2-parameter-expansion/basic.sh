
${parameter:-word}

${parameter-word}

${parameter:=word}

${parameter=word}

${parameter:?word}

${parameter?word}

${parameter:+word}

${parameter+word}
