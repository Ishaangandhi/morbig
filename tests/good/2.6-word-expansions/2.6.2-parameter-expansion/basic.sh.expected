[
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName",
                [
                  "CmdName_Word",
                  [
                    "Word",
                    "${parameter-word}",
                    [
                      [
                        "WordVariable",
                        [
                          "VariableAtom",
                          "parameter",
                          [
                            "UseDefaultValues",
                            [
                              "Word",
                              "word",
                              [
                                [
                                  "WordDoubleQuoted",
                                  [
                                    "Word",
                                    "word",
                                    [ [ "WordLiteral", "word" ] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName",
                [
                  "CmdName_Word",
                  [
                    "Word",
                    "${parameter-word}",
                    [
                      [
                        "WordVariable",
                        [
                          "VariableAtom",
                          "parameter",
                          [
                            "UseDefaultValues",
                            [
                              "Word",
                              "word",
                              [
                                [
                                  "WordDoubleQuoted",
                                  [
                                    "Word",
                                    "word",
                                    [ [ "WordLiteral", "word" ] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName",
                [
                  "CmdName_Word",
                  [
                    "Word",
                    "${parameter=word}",
                    [
                      [
                        "WordVariable",
                        [
                          "VariableAtom",
                          "parameter",
                          [
                            "AssignDefaultValues",
                            [
                              "Word",
                              "word",
                              [
                                [
                                  "WordDoubleQuoted",
                                  [
                                    "Word",
                                    "word",
                                    [ [ "WordLiteral", "word" ] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName",
                [
                  "CmdName_Word",
                  [
                    "Word",
                    "${parameter=word}",
                    [
                      [
                        "WordVariable",
                        [
                          "VariableAtom",
                          "parameter",
                          [
                            "AssignDefaultValues",
                            [
                              "Word",
                              "word",
                              [
                                [
                                  "WordDoubleQuoted",
                                  [
                                    "Word",
                                    "word",
                                    [ [ "WordLiteral", "word" ] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName",
                [
                  "CmdName_Word",
                  [
                    "Word",
                    "${parameter?word}",
                    [
                      [
                        "WordVariable",
                        [
                          "VariableAtom",
                          "parameter",
                          [
                            "IndicateErrorifNullorUnset",
                            [
                              "Word",
                              "word",
                              [
                                [
                                  "WordDoubleQuoted",
                                  [
                                    "Word",
                                    "word",
                                    [ [ "WordLiteral", "word" ] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName",
                [
                  "CmdName_Word",
                  [
                    "Word",
                    "${parameter?word}",
                    [
                      [
                        "WordVariable",
                        [
                          "VariableAtom",
                          "parameter",
                          [
                            "IndicateErrorifNullorUnset",
                            [
                              "Word",
                              "word",
                              [
                                [
                                  "WordDoubleQuoted",
                                  [
                                    "Word",
                                    "word",
                                    [ [ "WordLiteral", "word" ] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName",
                [
                  "CmdName_Word",
                  [
                    "Word",
                    "${parameter+word}",
                    [
                      [
                        "WordVariable",
                        [
                          "VariableAtom",
                          "parameter",
                          [
                            "UseAlternativeValue",
                            [
                              "Word",
                              "word",
                              [
                                [
                                  "WordDoubleQuoted",
                                  [
                                    "Word",
                                    "word",
                                    [ [ "WordLiteral", "word" ] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName",
                [
                  "CmdName_Word",
                  [
                    "Word",
                    "${parameter+word}",
                    [
                      [
                        "WordVariable",
                        [
                          "VariableAtom",
                          "parameter",
                          [
                            "UseAlternativeValue",
                            [
                              "Word",
                              "word",
                              [
                                [
                                  "WordDoubleQuoted",
                                  [
                                    "Word",
                                    "word",
                                    [ [ "WordLiteral", "word" ] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ]
]